frontend/online-shop/src/components/ProductDetails.tsx

```diff
@@ -0,0 +1,60 @@
+import {
+  Button,
+  Card,
+  CardContent,
+  CardHeader,
+  CardMedia,
+  Grid,
+  Typography,
+} from "@material-ui/core";
+import { Link, useParams } from "react-router-dom";
+import ShoppingCartIcon from "@material-ui/icons/ShoppingCart";
+import { makeStyles } from "@material-ui/core/styles";
+
+const useStyles = makeStyles((theme) => ({
+  customGrid: {
+    justifyContent: "center",
+  },
+  customText: {
+    color: theme.palette.primary.main,
+  },
+  customCard: {
+    borderStyle: "none",
+    borderRadius: 3,
+  },
+}));
+
+function ProductDetails(props: any) {
+  // const params = useParams()
+  // return (<h1>Hello {(params as any).productId} and {product.name}</h1>)
+  const classes = useStyles();
+  const product = props.location.state.currentProduct;
```

Comment:
As this is a component that is directly tied to a route you shouldn't use props with it. Indeed when you come to this page from the products list page you will be able to send the product and retrieve it from the props. How about when you try to access directly http://localhost:4200/products/1? Also there is a method called useParams() that can retrieve the path parameters object without using props

Reply:
You should retrieve the product id from the path and make a service call to retrieve the product with the respective id.